### PR TITLE
fix: evaluate ignoreChanges before checking if pre-submit should always run

### PR DESCRIPTION
**Pull request by ankitm123**
Signed-off-by: ankitm123 <ankitmohapatra123@gmail.com>

ignore changes are not ignored if always_run is set to true.
So even if the user wants to not trigger a build when README file is changed for example, lighthouse will still trigger a build because always_run is evaluated first in the logic.

Context: https://kubernetes.slack.com/archives/C9MBGQJRH/p1657794048005119

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,31 +36,31 @@ build: build-webhooks build-poller build-keeper build-foghorn build-tekton-contr
 
 .PHONY: build-webhooks
 build-webhooks: ## Build the webhooks controller binary for the native OS
-	$(GO) build -i -ldflags "$(GO_LDFLAGS)" -o bin/$(WEBHOOKS_EXECUTABLE) $(WEBHOOKS_MAIN_SRC_FILE)
```
Comment (ankitm123) [inline Makefile:39]: `-i` is deprecated in go

**Comment by ankitm123** [comment]
/hold 
going to add a test case for this, so we dont regress in the future.

**Review by rajatgupta24 — APPROVED** [review]
lgtm

**Pull request merged**